sklearn/utils/_plotting.py

```diff
@@ -63,6 +69,218 @@ def _validate_from_predictions_params(
 
         return pos_label, name
 
+    @classmethod
+    def _validate_from_cv_results_params(
+        cls,
+        cv_results,
+        X,
+        y,
+        *,
+        sample_weight,
+        pos_label,
+        # name,
+        # curve_kwargs,
+    ):
+        check_matplotlib_support(f"{cls.__name__}.from_cv_results")
+
+        required_keys = {"estimator", "indices"}
+        if not all(key in cv_results for key in required_keys):
+            raise ValueError(
+                "`cv_results` does not contain one of the following required keys: "
+                f"{required_keys}. Set explicitly the parameters "
+                "`return_estimator=True` and `return_indices=True` to the function"
+                "`cross_validate`."
+            )
+
+        train_size, test_size = (
+            len(cv_results["indices"]["train"][0]),
+            len(cv_results["indices"]["test"][0]),
+        )
+
+        if _num_samples(X) != train_size + test_size:
+            raise ValueError(
+                "`X` does not contain the correct number of samples. "
+                f"Expected {train_size + test_size}, got {_num_samples(X)}."
+            )
+
+        if type_of_target(y) != "binary":
+            raise ValueError(
+                f"The target `y` is not binary. Got {type_of_target(y)} type of"
+                " target."
+            )
+        check_consistent_length(X, y, sample_weight)
+
+        try:
+            pos_label = _check_pos_label_consistency(pos_label, y)
+        except ValueError as e:
+            # Adapt error message
+            raise ValueError(str(e).replace("y_true", "y"))
+
+        # n_curves = len(cv_results["estimator"])
+        # # NB: Both these also checked in `plot`, but thought it best to fail earlier.
+        # cls._validate_multi_curve_kwargs(cls, curve_kwargs, name, n_curves)
+        # if isinstance(name, list) and len(name) not in (1, n_curves):
+        #     raise ValueError(
+        #         f"`name` must be None, a list of length {n_curves} or a single "
+        #         f"string. Got list of length: {len(name)}."
+        #     )
```

Comment:
I felt that we should 'fail earlier' for this check and also the `**kwarg` deprecation (which is done in `_validate_curve_kwargs`, which not run until `plot`. In some cases we do a bit of compute before these 2 checks are run. BUT doing this checking earlier does result in redundant validation as everything is eventually passed to `plot`. 

WDYT @jeremiedbb @glemaitre

Reply:
After discussing with @glemaitre , we decided that while it would be nice to fail earlier, it is okay here that the validation is delayed until `plot`.

Ultimately, it is due to how Displays are implemented. Potentially if we had implemented it such that the call to plot was separate, e.g.,
```python
display = RocCurveDisplay.from_estimator(kwargs)
display.plot(kwargs)
```

Then the plot kwargs could be validated at `plot`, but oh well!